### DISCUSSION TITLE
<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#34221] [Help]
### Run `next info` (available from version 12.0.8 and up)

<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed

### What version of Next.js are you using?

12.0

### What version of Node.js are you using?

14

### What browser are you using?

chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

‘<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed
Remove window.html __NEXT_DATA__ variable to avoid a large amount of data in HTML affecting SEO effect’

### Expected Behavior

‘<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed
Remove window.html __NEXT_DATA__ variable to avoid a large amount of data in HTML affecting SEO effect’

### To Reproduce

‘<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed
Remove window.html __NEXT_DATA__ variable to avoid a large amount of data in HTML affecting SEO effect’

**Comment by parlay96** [comment]
It seriously affects SEO and must be removed

**Comment by parlay96** [comment]
‘<script id="__NEXT_DATA__" type="application/json"> It seriously affects SEO and must be removed Remove window.html __NEXT_DATA__ variable to avoid a large amount of data in HTML affecting SEO effect’

**Comment by parlay96** [comment]
https://umijs.org/zh-CN/config#ssr   This is China's SSR technology scheme. They can delete props data

**Comment by Bahlaouane-Hamza** [comment]
Hi,
You will need to patch Next.js for this, here's one for next@12.1.6
You can use https://www.npmjs.com/package/patch-package to auto install the patch.

[next+12.1.6.patch](https://github.com/vercel/next.js/files/11009343/next%2B12.1.6.patch)

For this part
```js
+    delete props1.pageProps.node
+    delete props1.pageProps.i18n
+    delete props1.pageProps.menus
```
I'm removing only the props i don't wanna keep in the HTML, If you are using NextAuth for example, you want to keep `session` coming from SSR, and this may vary depending on your use case.

And finally on the _hydrate part, i just restore the props i removed.

Hope this helps.
